### PR TITLE
Feature: low gas claim comp

### PR DESCRIPTION
**Pull request by shmuel-web**
Feature: low gas claim comp

Screen Shot 2021-06-23 at 11 17 49
Screen Shot 2021-06-23 at 11 16 23
Screen Shot 2021-06-23 at 11 17 31

### CODE REVIEW

File: src/components/modals/ClaimCompModal.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+      this.props.efficientClaim()
+    } else {
+      this.props.fullClaim()
+    }
+    EventBus.$emit("close-modal")
+  }
+
+  render() {
+    const {avatarCompBalance, compBalance, efficientCompClaimGasEstimate, compClaimGasEstimate} = compoundStore
+    const efficientClaimPrecentage = parseInt(percentage(avatarCompBalance, compBalance))
```
Comment (yaronvel) [inline src/components/modals/ClaimCompModal.js:121]: big numbers?

File: src/components/modals/ClaimCompModal.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                </Flex>
+                <Flex style={{padding: "22px"}} justifyBetween>
+                  <Text>Gas</Text>
+                  <Text>{displayNum(compClaimGasEstimate, 4)} ETH</Text>
+                </Flex>
+            </ContentBox>
+            <ExplainerText>
+              Full calim process that will retrive 100% of COMP but with less gas efficiency.
+            </ExplainerText>
+            <ModalButton onClick={() => this.claim()}>
```
Comment (yaronvel) [inline src/components/modals/ClaimCompModal.js:160]: don't be cheap. put `false` in the input parameter.

File: src/lib/Utils.js [hunk 3]
```diff
@@ -38,6 +38,8 @@ export const isKovan = () => {
 
 export const toCommmSepratedString = (n) => n.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
 
+export const percentage = (partialValue, totalValue) =>  (100 * partialValue) / totalValue;
```
Comment (yaronvel) [inline src/lib/Utils.js:41]: maybe do the big number translation here?

File: src/lib/Utils.js [hunk 4]
```diff
@@ -65,4 +67,10 @@ export const fromUiDeciamlPointFormat = (num, decimalPoint) => {
     const x = toBN(toWei(num))
     return x.div(factor)
 }
-  
\ No newline at end of file
+  
+export const gasToEth = async (gas, web3) => {
+    const gasPrice = await web3.eth.getGasPrice()
+    const cost = gas * gasPrice
```
Comment (yaronvel) [inline src/lib/Utils.js:73]: do we need big num math here?

**Review by yaronvel — CHANGES_REQUESTED** [review]

File: src/lib/Utils.js (hunk 4 above)
Comment (shmuel-web) [reply src/lib/Utils.js:73]: gas is in Wei
and gas price is in Gwei (8 numbers after the decimal)
these are pretty small numbers

and I compared them to the calculation meta mask is making...

File: src/components/modals/ClaimCompModal.js (hunk 1 above)
Comment (shmuel-web) [reply src/components/modals/ClaimCompModal.js:121]: they are floats using fromWei

File: src/lib/Utils.js (hunk 3 above)
Comment (shmuel-web) [reply src/lib/Utils.js:41]: no need  
we
are calculating a percentage of normalized numbers

File: src/components/modals/ClaimCompModal.js (hunk 2 above)
Comment (shmuel-web) [reply src/components/modals/ClaimCompModal.js]: done!

**Review by yaronvel — APPROVED** [review]

**Pull request merged**